File: Javascript/Algorithms/1476.js
var SubrectangleQueries = function(rectangle) {
    this.rect = rectangle;    
};

SubrectangleQueries.prototype.updateSubrectangle = function(row1, col1, row2, col2, newValue) {
    const rect = this.rect;
    for (let i = row1; i <= row2; i++)
        for (let j = col1; j <= col2; j++)
            rect[i][j] = newValue;
};

SubrectangleQueries.prototype.getValue = function(row, col) {
    return this.rect[row][col];
};

File: Python/Algorithms/1476.py
class SubrectangleQueries:
    def __init__(self, rectangle):
        self.rect = rectangle

    def updateSubrectangle(self, row1, col1, row2, col2, newValue):
        rect = self.rect
        rows, cols = range(row1, row2 + 1), range(col1, col2 + 1)
        for i in rows:
            r = rect[i]
            for j in cols:
                r[j] = newValue

    def getValue(self, row, col):
        return self.rect[row][col]
